--- a/fs/readdir.c
+++ b/fs/readdir.c
@@ -340,9 +340,10 @@
 #ifdef CONFIG_ZEROMOUNT
 skip_real_iterate:
 	if (error >= 0 && !signal_pending(current)) {
 		zeromount_inject_dents(f.file, (void __user **)&dirent, &count, &f.file->f_pos);
-		if (count != initial_count)
+		if (count != initial_count) {
 			error = initial_count - count;
-		goto zm_out;
+			goto zm_out;
+		}
 	}
 #endif
 	if (buf.prev_reclen) {
@@ -468,9 +469,10 @@
 #ifdef CONFIG_ZEROMOUNT
 skip_real_iterate:
 	if (error >= 0 && !signal_pending(current)) {
 		zeromount_inject_dents64(f.file, (void __user **)&dirent, &count, &f.file->f_pos);
-		if (count != initial_count)
+		if (count != initial_count) {
 			error = initial_count - count;
-		goto zm_out;
+			goto zm_out;
+		}
 	}
 #endif
 	if (buf.prev_reclen) {
@@ -701,9 +703,10 @@
 #ifdef CONFIG_ZEROMOUNT
 skip_real_iterate:
 	if (error >= 0 && !signal_pending(current)) {
 		zeromount_inject_dents(f.file, (void __user **)&dirent, &count, &f.file->f_pos);
-		if (count != initial_count)
+		if (count != initial_count) {
 			error = initial_count - count;
-		goto zm_out;
+			goto zm_out;
+		}
 	}
 #endif
 	if (buf.prev_reclen) {
